File: internal/models/member.go
package models

// Member is a user's membership in a organization, with their organization-scoped role (if
// any). Flattened for the members list.
type Member struct {
	UserID   string `json:"user_id"`
	Email    string `json:"email"`
	FullName string `json:"full_name"`
	Role     string `json:"role"` // organization-scoped role key, or "" if none
}

File: web/src/types/member.ts
/** A organization member with their organization-scoped role (if any). */
export interface Member {
  user_id: string
  email: string
  full_name: string
  role: string
}

/** Payload to add a member to a organization. */
export interface AddMemberInput {
  email: string
  role?: string
}
